File: app/src/main/java/com/yyhd/myreader/db/DBManager.java
//package com.yyhd.myreader.db;
//
//import android.content.Context;
//import android.database.sqlite.SQLiteDatabase;
//
//
//
///**
// * Created by hanli
// * date 2019/4/23.
// * ps: 数据库的管理者，单例形式
// */
//public class DBManager {
//
//    /**
//     * 单例对象
//     */
//    private static DBManager mDBManager = new DBManager();
//
//    /**
//     * 数据库的连接
//     */
//    private DaoSession mDaoSession;
//
//    private DBManager(){
//    }
//
//    public static DBManager getInstance(){
//        if(mDBManager == null) {
//            mDBManager = new DBManager();
//        }
//        return mDBManager;
//    }
//
//    /**
//     * 初始化greenDao数据库相关
//     */
//    public void initGreenDao(Context context){
//        DaoMaster.DevOpenHelper helper = new MyOpenHelper(context, "yyhd");
//        SQLiteDatabase writableDatabase =helper.getWritableDatabase();
//        DaoMaster master = new DaoMaster(writableDatabase);
//        mDaoSession = master.newSession();
//    }
//
//
//    /**
//     * 获得数据库的daoSession
//     * @return
//     */
//    public DaoSession getDaoSession(){
//        return mDaoSession;
//    }
//}

File: app/src/main/java/com/yyhd/myreader/db/DBManager.kt
package com.yyhd.myreader.db

import android.content.Context
import com.yyhd.myreader.db.table.DaoMaster
import com.yyhd.myreader.db.table.DaoSession

/**
 * Created by hanli
 * date 2021-03-01.
 * ps:
 */
object DBManager {

    open lateinit var mDaoSession : DaoSession

    fun initGreenDao(context: Context){
        val helper = MyOpenHelper(context, "hanli_book")
        val writableDatabase = helper.getWritableDatabase()
        val master = DaoMaster(writableDatabase)
        mDaoSession = master.newSession()
    }

}